routers/web/repo/setting.go

```diff
@@ -892,6 +892,16 @@ func CollaborationPost(ctx *context.Context) {
 		return
 	}
 
+	// find the owner team of the organization the repo belongs too and
+	// check if the user we're trying to add is an owner.
+	teams, err := organization.GetRepoTeams(ctx, ctx.Repo.Repository)
+	for _, team := range teams {
+		if team.IsOwnerTeam() && team.IsMember(u.ID) {
+			ctx.Redirect(setting.AppSubURL + ctx.Req.URL.EscapedPath())
+			return
+		}
+	}
+
```

Comment:
I think you should move this below line 922 and you should check that ctx.Repo.Repository.Owner.IsOrganization()

Reply:
applied and also missing error check checked :)